src/openassetio-core/tests/hostApi/ManagerTest.cpp

```diff
@@ -413,3 +413,342 @@ TEMPLATE_TEST_CASE(
+                   preflight(refs, traits, context, hostSessionPtr, _, _))
+          .LR_SIDE_EFFECT(_5(0, expectedVec[0]))
+          .LR_SIDE_EFFECT(_5(1, expectedVec[1]))
+          .LR_SIDE_EFFECT(_5(2, expectedVec[2]));
+
+      WHEN("batch preflight is called with default errorPolicyTag") {
+        const std::vector<openassetio::EntityReference> actualVec =
+            manager->preflight(refs, traits, context);
+        THEN("returned list of TraitsDatas is as expected") { CHECK(expectedVec == actualVec); }
+      }
+      WHEN("batch preflight is called with kException errorPolicyTag") {
+        const std::vector<openassetio::EntityReference> actualVec = manager->preflight(
+            refs, traits, context, hostApi::Manager::BatchElementErrorPolicyTag::kException);
+        THEN("returned list of TraitsDatas is as expected") { CHECK(expectedVec == actualVec); }
+      }
+      WHEN("batch preflight is called with kVariant errorPolicyTag") {
+        std::vector<std::variant<openassetio::BatchElementError, openassetio::EntityReference>>
+            actualVec = manager->preflight(refs, traits, context,
+                                           hostApi::Manager::BatchElementErrorPolicyTag::kVariant);
+        THEN("returned lists of variants contains the expected TraitsDatas") {
+          CHECK(expectedVec.size() == actualVec.size());
+          for (size_t i = 0; i < actualVec.size(); ++i) {
+            CHECK(std::holds_alternative<openassetio::EntityReference>(actualVec[i]));
+            auto actualVal = std::get<openassetio::EntityReference>(actualVec[i]);
+            CHECK(expectedVec[i] == actualVal);
+          }
+        }
+      }
+    }
+    GIVEN(
+        "manager plugin successfully preflights multiple entity references in a non-index order") {
+      const openassetio::EntityReferences refs = {openassetio::EntityReference{"testReference1"},
+                                                  openassetio::EntityReference{"testReference2"},
+                                                  openassetio::EntityReference{"testReference3"}};
+
+      const openassetio::EntityReference expected1{"ref1"};
+      const openassetio::EntityReference expected2{"ref2"};
+      const openassetio::EntityReference expected3{"ref3"};
+      std::vector<openassetio::EntityReference> expectedVec{expected1, expected2, expected3};
+
+      // With success callback side effect, given out of order.
+      REQUIRE_CALL(*mockManagerInterfacePtr,
+                   preflight(refs, traits, context, hostSessionPtr, _, _))
+          .LR_SIDE_EFFECT(_5(2, expectedVec[2]))
+          .LR_SIDE_EFFECT(_5(0, expectedVec[0]))
+          .LR_SIDE_EFFECT(_5(1, expectedVec[1]));
+
+      WHEN("batch preflight is called with default errorPolicyTag") {
+        const std::vector<openassetio::EntityReference> actualVec =
+            manager->preflight(refs, traits, context);
+        THEN("returned list of TraitsDatas is ordered in index order") {
```

Comment:
[fix] As above, `list of EntityReferences`

Reply:
Good spot, fixed as part of b98670a7